File: Objective-C/LRI/Drink MVP/Model/DrinkModel.m
//
//  DrinkModel.m
//  LRI
//
//  Created by Development on 05.07.2022.
//

#import <Foundation/Foundation.h>
#import "DrinkModel.h"

@interface DrinkModel ()
@property (nonatomic, strong) NSMutableArray<DrinkItem *> *items;
@end

@implementation DrinkModel

- (instancetype)init {
    self = [super init];
    if (self) {
        self.items = [[NSMutableArray alloc] init];
    }
    return self;
}

- (instancetype)initWithItems:(NSMutableArray<DrinkItem *> *) items {
    self = [super init];
    if (self) {
        self.items = items;
    }
    return self;
}

- (void)fillDataWith:(NSMutableArray<DrinkItem *> *) items {
    self.items = items;
}

- (NSMutableArray<DrinkItem *> *)getItems {
    return self.items;
}

- (NSInteger)getCount {
    return [self.items count];
}

@end

File: SwiftUIKit/LRI/Drink MVP/Model/DrinkModel.swift
//
//  DrinkModel.swift
//  LRI
//
//  Created by Development on 02.07.2022.
//

import Foundation

protocol DrinkModelDataAccess {
    var items: [DrinkItem] { get }
    var count: Int { get }
}

class DrinkModel: DrinkModelDataAccess {
    private(set) var items: [DrinkItem] = []
    var count: Int { items.count }
    
    init(items: [DrinkItem] = []) {
        self.items = items
    }
}

extension DrinkModel {
    func fillData(with items: [DrinkItem]) {
        self.items = items
    }
}
